azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -284,22 +297,26 @@ public Set<Integer> getContainersOfTerminatedFlows() {
   }
 
   /**
-   * Get the flows of EXECUTION_STOPPED status
+   * Get the flows recently (last 15 minutes) terminated: status in (execution_stopped, killed,
+   * killing, failed, failed_finishing, failed_succeeded, canceled)
    */
   @NotNull
-  Set<Integer> getExecutionStoppedFlows() {
-    Set<Integer> executionStoppedFlows = new HashSet<>();
-    if (this.executionStoppedMap.containsKey(Status.EXECUTION_STOPPED)) {
+  Set<Integer> getRecentlyTerminatedFlows() {
+    Set<Integer> recentlyTerminatedFlows = new HashSet<>();
+    this.recentTerminatedStatusMap.forEach((status, value) -> {
       try {
         List<ExecutableFlow> flows = this.executorLoader.fetchFreshFlowsForStatus(
-            Status.EXECUTION_STOPPED, this.executionStoppedMap);
-        executionStoppedFlows.addAll(
+            status, this.recentTerminatedStatusMap);
+        recentlyTerminatedFlows.addAll(
             flows.stream().map(ExecutableFlow::getExecutionId).collect(Collectors.toSet()));
+        logger.info("Got recently terminated flows executions of Status " + status + ": " +
+            flows.stream().map(ExecutableFlow::getExecutionId).map(Objects::toString)
+                .collect(Collectors.joining(",")));
```

Comment:
Nit: Consider using a StringBuffer for this, as currently implemented it will result in a larger number of string objects created than needed.

Reply:
Sure